src/Reflection/ReflectionClass.php

```diff
@@ -282,7 +283,25 @@ private function getParentMethods(): array
         return array_merge(
             [],
             ...array_map(
-                static fn (ReflectionClass $ancestor): array => $ancestor->getMethods(),
```

Comment:
I'm a bit puzzled about this: why did `$ancestor->getMethods()` not suffice? We're kinda exploding and re-assembling everything here? :thinking:

Reply:
I need to pass `$this` as `currentClass` to parent methods so I can resolve `static` in `ReflectionMethod`. Yeah, it's not perfect but same pattern is used for trait methods as well. There may be better way how to solve it but it would need more refactoring of `getMethods()` etc.